### PR TITLE
Add detect_restart_file

### PR DESCRIPTION
**Pull request by Sbozzolo**
As more downstream packages are dealing with restart files, I started moving some shared functionalities to ClimaUtilities.

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -4,21 +4,28 @@ ClimaUtilities.jl Release Notes
...
 as much. More redundant allocations due to `Utils.isequispaced` were fixed.
 
+### Features
+
+#### Added `OutputPathGenerator.detect_restart_file`. PR [#105](https://github.com/CliMA/ClimaUtilities.jl/pull/120)
+
+A new function, `OutputPathGenerator.detect_restart_file`, has been added to
+help automatically locate the most recent restart file generated by a
+simulation. This function is particularly useful when restarting simulations
+from a previous checkpoint. It works with the `ActiveLinkStyle` can be
```
Comment (juliasloan25) [inline NEWS.md:19]:
```suggestion
from a previous checkpoint. It works with the `ActiveLinkStyle` and can be
```

**Review by juliasloan25 — APPROVED** [review]
Thank you for the thorough documentation!

Btw, the docstring for [generate_output_path](https://github.com/CliMA/ClimaUtilities.jl/blob/2c796a7db6321c65b4f4a66b3e4872cdef7d04b7/src/OutputPathGenerator.jl#L106) is out of date - could you fix it in this PR please?

File: src/OutputPathGenerator.jl [hunk 2]
```diff
@@ -162,9 +164,6 @@ end
 Documentation for this function is in the `ActiveLinkStyle` struct.
 """
 function generate_output_path(::ActiveLinkStyle, output_path; context = nothing)
-    # TODO: At the moment we hard code the name to be something like output_0000. We could make
-    # this customizable as an attribute in ActiveLinkStyle.
```
[the remark this answers is not in the document]
Comment (Sbozzolo) [reply src/OutputPathGenerator.jl:166]: It is, but at this point I don't want to make it customizable, otherwise lots of code would need to be changed.

**Comment by Sbozzolo** [comment]
> Thank you for the thorough documentation!
> 
> Btw, the docstring for [generate_output_path](https://github.com/CliMA/ClimaUtilities.jl/blob/2c796a7db6321c65b4f4a66b3e4872cdef7d04b7/src/OutputPathGenerator.jl#L106) is out of date - could you fix it in this PR please?

I wouldn't say that it is outdated. The docstring is telling users what type is expected, but I don't add the type restriction in the actual function because such restriction is not strictly needed. (There is no reason for the docstring to match the function signature: docstrings are meant to be for humans)

But I'll add the type restriction to make it more clear.

**Comment by juliasloan25** [comment]
> > Thank you for the thorough documentation!
> > Btw, the docstring for [generate_output_path](https://github.com/CliMA/ClimaUtilities.jl/blob/2c796a7db6321c65b4f4a66b3e4872cdef7d04b7/src/OutputPathGenerator.jl#L106) is out of date - could you fix it in this PR please?
> 
> I wouldn't say that it is outdated. The docstring is telling users what type is expected, but I don't add the type restriction in the actual function because such restriction is not strictly needed. (There is no reason for the docstring to match the function signature: docstrings are meant to be for humans)
> 
> But I'll add the type restriction to make it more clear.

I agree that the docstring and function don't need to match 100%. What I meant is that to me, `style::OutputPathGeneratorStyle = ActiveLinkStyle()` implies `ActiveLinkStyle <: OutputPathGeneratorStyle`, which doesn't seem to be true

**Comment by Sbozzolo** [comment]
> > > Thank you for the thorough documentation!
> > > Btw, the docstring for [generate_output_path](https://github.com/CliMA/ClimaUtilities.jl/blob/2c796a7db6321c65b4f4a66b3e4872cdef7d04b7/src/OutputPathGenerator.jl#L106) is out of date - could you fix it in this PR please?
> > 
> > 
> > I wouldn't say that it is outdated. The docstring is telling users what type is expected, but I don't add the type restriction in the actual function because such restriction is not strictly needed. (There is no reason for the docstring to match the function signature: docstrings are meant to be for humans)
> > But I'll add the type restriction to make it more clear.
> 
> I agree that the docstring and function don't need to match 100%. What I meant is that to me, `style::OutputPathGeneratorStyle = ActiveLinkStyle()` implies `ActiveLinkStyle <: OutputPathGeneratorStyle`, which doesn't seem to be true

Indeed, that's a bug! Thanks for catching that!

**Comment by Sbozzolo** [comment]
I also bumped to 0.1.16. We can merge this after #119

**Pull request merged**